src/Orchard.ContentManagement.Abstractions/ContentElement.cs

```diff
@@ -176,6 +176,13 @@ public void Weld(string name, object obj)
             {
                 Data[name] = JObject.FromObject(obj);
             }
+
+            // Also copy properties added to .Content
+            var contentPart = obj as ContentElement;
+            if (obj != null)
+            {
+                ((JObject)Data[name]).Merge(contentPart.Data, new JsonMergeSettings { MergeArrayHandling = MergeArrayHandling.Merge });
+            }
```

Comment:
Thanks, taking a look. Too messy to deal between parts and dynamic data right now ...